### PR TITLE
Group and Event integration

### PR DESCRIPTION
**Pull request by napalmpapalam**

### CODE REVIEW

File: src/proxy/src/calls/generic_calls.rs [hunk 1]
```diff
@@ -23,10 +21,10 @@ use ic_cdk::{
 };
 
 #[post_upgrade]
-pub fn post_upgrade() {
+pub async fn post_upgrade() {
     Websocket::init();
     RewardTimerStore::start();
-    BoostCalls::start_timers_after_upgrade();
+    let _ = BoostCalls::start_timers_after_upgrade().await;
```
Comment (napalmpapalam) [inline src/proxy/src/calls/generic_calls.rs:27]: not sure how to handle result here, do I need trap?

File: src/proxy/src/calls/group_calls.rs [hunk 2]
```diff
@@ -250,8 +252,8 @@ pub async fn remove_group_role(group_id: u64, role_name: String) -> CanisterResu
 #[update(guard = "is_not_anonymous")]
```
Comment (napalmpapalam) [inline src/proxy/src/calls/group_calls.rs]: why its update call?

File: src/proxy/src/calls/stats_calls.rs [hunk 3]
```diff
@@ -8,18 +8,13 @@ async fn store_stats() -> CanisterResult<Vec<String>> {
...
         format!("FriendRequestStore: {}", FriendRequestStore::size()),
-        format!("GroupStore: {}", GroupStore::size()),
-        format!("MemberStore: {}", MemberStore::size()),
-        format!("EventStore: {}", EventStore::size()),
-        format!("AttendeeStore: {}", AttendeeStore::size()),
+        format!("GroupStore: {}", groups().size().await?),
+        format!("EventStore: {}", events().size().await?),
         format!("ReportStore: {}", reports().size().await?),
-        format!("BoostStore: {}", BoostedStore::size()),
+        format!("BoostStore: {}", boosteds().size().await?),
```
Comment (rem-code-s) [inline src/proxy/src/calls/stats_calls.rs]: `boosteds`? 🤣  `boosts` is probably better

File: src/proxy/src/logic/report_logic.rs [hunk 4]
```diff
@@ -6,17 +6,22 @@ use catalyze_shared::{
...
 
 pub struct ReportCalls;
 
 impl ReportCalls {
     pub async fn add_report(post_report: PostReport) -> CanisterResult<ReportResponse> {
-        let (_, member) = MemberStore::get(caller())?;
+        let (_, group) = groups().get(post_report.group_id).await?;
 
-        if !member.is_group_joined(&post_report.group_id) {
+        let is_group_joined = group
```
Comment (rem-code-s) [inline src/proxy/src/logic/report_logic.rs]: There is also a `is_member` method

File: src/proxy/src/calls/generic_calls.rs (hunk 1 above)
Comment (rem-code-s) [reply src/proxy/src/calls/generic_calls.rs:27]: good question, i don't think its wise to trap on this on the post upgrade based the the importance of this timer, so i suggest leaving it as is

**Review by rem-code-s — CHANGES_REQUESTED** [review]

File: src/proxy/src/logic/report_logic.rs (hunk 4 above)
Comment (napalmpapalam) [reply src/proxy/src/logic/report_logic.rs]: Yeah, coz I added some after dat change :D

File: src/proxy/src/calls/group_calls.rs (hunk 2 above)
Comment (rem-code-s) [reply src/proxy/src/calls/group_calls.rs]: Seems like a mistake, the description says it should be a query

Comment (napalmpapalam) [reply src/proxy/src/calls/group_calls.rs]: fixed

**Review by rem-code-s — APPROVED** [review]

**Pull request merged**